patroni/postgresql/config.py

```diff
@@ -335,6 +336,9 @@ def __init__(self, postgresql, config):
         self._auto_conf = os.path.join(postgresql.data_dir, 'postgresql.auto.conf')
         self._auto_conf_mtime = None
         self._pgpass = config.get('pgpass') or os.path.join(os.path.expanduser('~'), 'pgpass')
+        if os.path.exists(self._pgpass) and not os.path.isfile(self._pgpass):
```

Comment:
It doesn't help much for the case when pgpass can't be created

Reply:
Should we try to create it from __init__ method?

Reply:
I think we should try to open it in append mode.
If the file can't be created or isn't writable open will fail.

Reply:
If we create file anyway, should we write credentials or keep an empty file?

Reply:
I added the directory of pgpassfile to check_directory which makes sure it's writable.